### PR TITLE
Chroot support

### PR DESCRIPTION
**Pull request by jreidinger**
tested manually only as it require root permissions for chrooting.

### CODE REVIEW

**Comment by jreidinger** [comment]
coverage decrease again caused by bad detection of code after fork

File: README.md [hunk 1]
```diff
@@ -226,6 +227,15 @@ Dir.chdir("/workspace") do
 end
 ```
 
+### Changing System Root
+
+If command have to be run on different system root, then :chroot options
```
Comment (lslezak) [inline README.md]: Um, I'd suggest better wording:

```
If a command needs to be executed in a different system root then the `:chroot` option can be used:
```

File: README.md [hunk 2]
```diff
@@ -42,6 +42,7 @@ Features
   * Thread-safety
   * Allows overriding environment variables
   * Optional logging for easy debugging
+  * Running on changed root ( require chroot permission )
```
Comment (lslezak) [inline README.md]: (it) require**s**

File: lib/cheetah.rb [hunk 3]
```diff
@@ -308,6 +309,8 @@ class << self
     #   @option options [Hash] :env ({})
     #     Allows to update ENV for the time of running the command. if key maps to nil value it
     #     is deleted from ENV.
+    #   @option options [String] :chroot ("/")
+    #     Allows to run on different system root.
```
Comment (lslezak) [inline lib/cheetah.rb]: What happens when `chroot` call fails? A documentation is missing...

Comment (jreidinger) [inline lib/cheetah.rb]: exception is raised and due to improvement in catching system error it is written to pipe, so user see it. For my testing script:

```
jreidinger@pepa:~/cheetah> ruby test.rb 
/home/jreidinger/cheetah/lib/cheetah.rb:641:in `check_errors': Execution of "touch /aaa" failed with status 127: Operation not permitted @ dir_s_chroot - /tmp/root. (Cheetah::ExecutionFailed)
	from /home/jreidinger/cheetah/lib/cheetah.rb:404:in `run'
	from test.rb:3:in `<main>'
```

Comment (jreidinger) [inline lib/cheetah.rb]: so it behave in same way like if fork or exec failed

**Comment by ancorgs** [comment]
LGTM

**Comment by lslezak** [comment]
LGTM, thanks!

**Pull request merged**